assets/js/components/KeyMetrics/MetricsSelectionPanel/Footer.js

```diff
@@ -141,49 +144,68 @@ export default function Footer( {
 		[ saveKeyMetricsSettings ]
 	);
 
-	const onSaveSuccess = useCallback( () => {
-		trackEvent( trackingCategory, 'metrics_sidebar_save' );
-
-		if ( isGA4Connected && hasMissingCustomDimensions ) {
-			setValues( FORM_CUSTOM_DIMENSIONS_CREATE, {
-				autoSubmit: true,
-			} );
+	const onSaveSuccess = useCallback(
+		( selectedItemSlugs ) => {
+			const userSavedConversionReportingKmwList = Object.values(
+				acrSpecificKmwWidgets
+			)
+				.flat()
+				.filter( ( n ) => selectedItemSlugs.includes( n ) );
+
+			// Include the conversion_reporting tracking label if necessary.
+			if ( userSavedConversionReportingKmwList ) {
```

Comment:
This condition will always return `true`. Since above you are using `filter` it will always result in an array, empty one or with the filtered items. So either include `userSavedConversionReportingKmwList.length` or use `.some` in the check above instead of `.filter` to return boolean value

Reply:
@zutigrm this has been addressed in my recent commits, thank you.